### ISSUE TITLE
EMG Training Module

### ISSUE THREAD

**Issue by satanb4** [satanb4/physionix#47]
# Add a training Module for the EMG data
A training module using a neural network would be a great addition for generating custom profiles for users.

- [x] Create a Neural Network
- [ ] Connect Neural Network to the main EMGData acquisition Class
- [ ] Parallelize or Optimize the training network with CUDA? (open for discussion)
- [ ] Run it as a separate, on-demand module and save its weights for each user as customized
- [ ] Integrate output into recognition algorithm for movements

This work would benefit the development of EMG based muscle movement classification for users with unique kinds of ailments.
PS - Initial Neural Network implementation from scratch done by @satanb4 already in "development" branch

**Comment by satanb4** [comment]
With testing data, getting accuracy of 99.95% now only on 25 epochs and 0.005 step size:-
![image](https://user-images.githubusercontent.com/26685910/236889621-c61caaab-dc36-49ca-a72c-661b9fc732c1.png)
OpenMP has been implemented. Would be nice to have CUDA accelerated processing, but my CodeSpaces device does not have the CUDA graphics. Maybe I can look into further optimization with parallel processing on CPU as well. Probably with more threads?
